### ISSUE TITLE
hook_config_info() needed for importing/exporting configurations

### ISSUE THREAD

**Issue by oriolroger** [backdrop-contrib/search_api_solr#5]
Hi, after changing a value in SOLR settings in a Backdrop site, when importing the configurations files (from the Configuration manager UI or using drush or bee), we got the sync error _"The configuration "search_api_solr.settings" is not owned by any module. Try enabling the module that provides this configuration, then importing again."_

It seems the search_api_solr settings are not found in the 'config_info'.

To fix this, we've added this hook in search_api_solr.module:

```
/**
 * Implements hook_config_info().
 */
function search_api_solr_config_info() {
  $prefixes['search_api_solr.settings'] = array(
    'label' => t('Search API SOLR settings'),
    'group' => t('Configuration'),
  );
  return $prefixes;
}
```

Maybe this should be added as a PR in the next version of the module?

Thanks.

**Comment by herbdool** [comment]
Thanks @oriolroger. I've added a PR based on that. Though I added a couple other fixes: added an update hook; removed the tasks variable since it's actually for `search_api_tasks`; removed the optimize variable since it should be a state variable like this `state_set('search_api_solr_last_optimize', REQUEST_TIME)`. Would you like to test?

We might be able to get the @backdrop-contrib/bug-squad to merge this after testing since the maintainer is unresponsive.

**Comment by laryn** [comment]
> since the maintainer is unresponsive.

FYI I received an email from the maintainer in regard to a separate module and he indicated he's withdrawn from the Backdrop project because of the focus of his day job and lack of extra personal time to devote to Backdrop.

**Comment by herbdool** [comment]
@laryn do you feel like taking this on too?

@argiepiano I mentioned checking in the update hook if the config is new but I realized it probably would never show it's new because the update hook runs after installation at which point the config file has already been copied over.

**Comment by laryn** [comment]
@herbdool I don't have a site using it currently so I don't think I will take this one at the moment.

**Comment by argiepiano** [comment]
> I mentioned checking in the update hook if the config is new but I realized it probably would never show it's new because the update hook runs after installation at which point the config file has already been copied over.

@herbdool, I think we still need to check if there is an existing configuration in the update hook. Consider the following scenario:

1. A user installed this module BEFORE there was any update hook (current state). The user has created custom configurations for the module through the UI
2. We add `search_api_solr_update_1000()` intended to move variables to config
3. This update WILL run on existing installations, since it did not exist before
4. When this update hook runs, it will basically move empty values (stored in variables) into those existing configs.

Because   `search_api_solr_update_1000()` does not exist in the current version, and because there are already some installations of the module, we need to check if those intallations already have a configuration set up.

**Comment by herbdool** [comment]
@argiepiano I get the scenario but I'm not sure how to check. I think `Config::isNew()` would always be false, since a newly installed module would have already created the config in a previous step, and already installed module would already have it too.

**Comment by argiepiano** [comment]
@herbdool, I've done some testing in upgrading a D7 site. I created a D7 custom module that created a variable. Then I created a parallel module in BAckdrop with the same name. The B module contained an upgrade hook to move the variable into config. ALSO, the B module contained a default configuration in the `config` folder with the same name as the config file created in the update hook.

 This is what I'm seeing:

1. When you upgrade from D7, the config file is NOT moved to `files/config_XXX` before the update hook is run
2. This means that you CAN check for `is_new` before moving the variables to config in the hook. This means that `Config::isNew()` will be TRUE in the upgrade hook when you upgrade from D7 for the first time.

Then, I tested a clean, vanilla installation of a Backdrop site. I modified the custom Backdrop module, so that when it was first installed it contained no update hook (mimicking what this module is doing). I manually created the config file and value. Then, after installing and after creating the config file, I created a hook_update_1000() to move a D7 variable to that config (which is what you are wanting to do here). So, adding this hook after the module was installed forced the site to run it (as expected), meaning that my manually created config setting was completely wiped out (which is what we don't want to happen).

To summarize, we should be check for `isNew()` for the config. This will result to TRUE during a D7 upgrade (meaning that it's safe to move the variable into config, as the site does not have that configuration file in `files/config`). But this will result in FALSE when this hook is run for an existing site (not an upgrade), avoiding the potential wiping out of previously set config values.

**Comment by herbdool** [comment]
Thanks for confirming that @argiepiano! I'll update the PR.

**Comment by argiepiano** [comment]
LGTM!
